LEGO1/mxstreamcontroller.cpp

```diff
@@ -133,6 +133,21 @@ MxResult MxStreamController::VTable0x30(MxDSAction* p_action)
 	return result;
 }
 
+// FUNCTION: LEGO1 0x100c1e70
+MxCore* MxStreamController::FUN_100c1e70(MxDSAction& p_obj)
+{
+	MxAutoLocker locker(&m_criticalSection);
+	MxCore* result = NULL;
+	if (p_obj.GetObjectId() != -1) {
+		MxDSAction* action = m_unk0x3c.Find(&p_obj, FALSE);
+		if (action != NULL) {
+			result = (MxCore*) action->GetUnknown28();
```

Comment:
Looks like we should change the type of `m_unk0x28` and the return type of this function to `MxPresenter*`? Based on how this function is used.